crates/core/tedge_mapper/src/operations.rs

```diff
@@ -0,0 +1,159 @@
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+};
+
+use crate::error::OperationsError;
+
+/// Operations are derived by reading files subdirectories per cloud /etc/tedge/operations directory
+/// Each operation is a file name in one of the subdirectories
+/// The file name is the operation name
+
+type Cloud = String;
+type OperationName = String;
+type Operation = HashMap<OperationName, PathBuf>;
+type OperationsMap = HashMap<Cloud, Operation>;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Operations {
+    dir: PathBuf,
+    operations: OperationsMap,
+}
+
+impl Default for Operations {
+    fn default() -> Self {
+        Self {
+            dir: Path::new("/etc/tedge/operations").to_path_buf(),
+            operations: HashMap::new(),
```

Comment:
This makes this struct a centralised repository of **all** operations **across clouds**. Do we want that? Or just have the individual cloud mappers keep only their own operation mappings? Not that I see any problem with this approach. Just asking.

Reply:
Is this default useful? I expect the mapper to call `Operations::try_new()` with the appropriate directory.